haystack/environment.py

```diff
@@ -32,10 +29,63 @@ def set_pytorch_secure_model_loading(flag_val="1"):
         logger.info("TORCH_FORCE_WEIGHTS_ONLY_LOAD is already set to %s, Haystack will use the same.", os_flag_val)
 
 
+def is_containerized() -> Optional[bool]:
```

Comment:
I've found https://github.com/ZiggerZZ/in_container/blob/master/in_container/in_container.py#L9 it's similar but it also looks for the .dockerenv file. Maybe we can incorporate that logic? (let's not depend on the package)